osxkeychain/osxkeychain_darwin.go

```diff
@@ -134,8 +134,21 @@ func (h Osxkeychain) List() (map[string]string, error) {
 	return resp, nil
 }
 
+// alwaysHTTPS takes a url and makes sure it always has the https protocol
+func alwaysHTTPS(url string) string {
```

Comment:
Actually it would be sort of a pain, we should verify that `url.Parse` rejects these right after.
A test for that would probably be better, just checking that `ftp://foobar:1234` fails